lib/Deserializer/functions.php

```diff
@@ -330,12 +330,12 @@ function mixedContent(Reader $reader): array
 }
 
 /**
- * The functionCaller deserializer turns an XML element into whatever your callable return.
+ * The functionCaller deserializer turns an XML element into whatever your callable returns.
  *
  * You can use, e.g., a named constructor (factory method) to create an object using
  * this function.
  *
- * @return mixed
+ * @return mixed whatever the 'func' callable returns
```

Comment:
I tried the above, and `phpstan` complains about the test code, for example:
```
------ ---------------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/Sabre/Xml/Deserializer/FunctionCallerTest.php                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  40     Parameter #2 $func of function Sabre\Xml\Deserializer\functionCaller expects callable(mixed ...): Sabre\Xml\Deserializer\Person,  
         array{'Sabre\\Xml\\Deserializer\\Person', 'fromXml'} given.                                                                       
         💡 Parameter #1 $name of passed callable is required but the parameter of accepting callable is optional. It might be called without it.                                                                                                                    
         💡 Parameter #2 $age of passed callable is required but accepting callable does not have that parameter. It will be called without it.                                                                                                                    
         💡 Parameter #3 $address of passed callable is required but accepting callable does not have that parameter. It will be called without it.                                                                                                                    
         💡 Parameter #4 $languages of passed callable is required but accepting callable does not have that parameter. It will be called without it.
```

I tried replacing `mixed...` with just `...` or `...mixed` or anything else I could think of. We want to say that the callable can have any number of parameters (including zero parameters) of any type - we do not care. The only thing we care about is that the callable passed in has some return type "CR" and the `functionCaller` returns whatever that "CR"  happens to be.

How to achieve this with PHP  doc and phpstan?

Reply:
the offical way atm seems to be

`* @param callable(mixed ...$params): T  $callable`

https://github.com/phpstan/phpstan/issues/5554#issuecomment-907821834

Reply:
I think we can scratch that for now

Reply:
> I think we can scratch that for now

I tried again whatever permutations I could think of, and no joy.